src/Symfony/Component/Translation/Translator.php

```diff
@@ -74,7 +74,7 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $configCacheFactory;
 
     /**
-     * @param string                         $locale    The locale
+     * @param string|null                    $locale    The locale
```

Comment:
This doesn't look right to me. If we were to make this change, we would also have to change the return type of `getLocale()` which would break the contract of the `TranslatorInterface`.

Reply:
I think the interface is lying and needs to be fixed. That is what I wrote in https://github.com/symfony/symfony/pull/33154#discussion_r315382160

Reply:
I don't see a use case for a `null` locale. My suggestion is to deprecate support for this instead (see #33272).

Reply:
Now that support for passing `null` is to be removed in Symfony 5.0 I would revert the docblock changes and only keep the check for `null` below.